test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs

```diff
@@ -140,6 +141,9 @@ void ConfigureTestServices(IServiceCollection services)
             Assert.Equal(3, activityProcessor.Invocations.Count); // begin and end was called
             var activity = (Activity)activityProcessor.Invocations[2].Arguments[0];
 
+            Assert.NotEmpty(activity.Tags.Where(tag => tag.Key == "enriched"));
+            Assert.Equal(shouldEnrich ? "yes" : "no", activity.Tags.Where(tag => tag.Key == "enriched").FirstOrDefault().Value);
```

Comment:
I think, this can be simplified to : 
if (shoudEnrich)
{
 assert that enriched tag exists and has the desired value.
}
else
{
assert that enriched tag don't exist
}

Reply:
That's definitely more elegant but I'm not sure if this would then still test what we were aiming for in #1549 . That issue asked for UTs for #1548 , which fixed specifically the order in which Enrich is called so that it is called after other tags are set, so we can check that it overrides any other modifiers to the activity. Therefore seems like we would need the tag to exist in both shouldEnrich = false and shouldEnrich = true , and have it be overwritten in one?